### ISSUE TITLE
upgrade to google play services 7

### ISSUE THREAD

**Issue by Bananeweizen** [cgeo/cgeo#4751]
We should have a look at using the google play services 7, and show the places settings dialog in case the gps is disabled (at least when invoking compass or map).

http://developer.android.com/google/play-services/index.html?utm_campaign=playservices65-1114&utm_source=dac&utm_medium=blog#

**Comment by Bananeweizen** [comment]
upgrade to play services is done, the settings dialog is not yet used

**Comment by Bananeweizen** [comment]
@samueltardieu Today I looked at this again. While I do understand the related code snippet in http://developer.android.com/reference/com/google/android/gms/location/SettingsApi.html, I currently don't get my head around how to connect that to our code. We initialize the location provider just using the application context, while that dialog related code seems to work only with an activity as context (which makes sense).

**Comment by samueltardieu** [comment]
Indeed. Without introducing the notion of "current activity" (which could be done quite easily since we use `AbstractActivity` as a base class and could register/unregister such an activity), we would have a hard time presenting such a dialog. But wouldn't this "current activity" information be a potential for abuse in the future?

**Comment by samueltardieu** [comment]
@Bananeweizen: With Google Play services 7.3, we get more information in the background on the availability of requested locations (https://developer.android.com/reference/com/google/android/gms/location/LocationCallback.html) which seems to have been added just for the reason I exposed above. From http://android-developers.blogspot.fr/2015/04/theres-lot-to-explore-with-google-play.html:

> For any ongoing location requests, it is important to know that you will continue to get good location data back. The SettingsApi is still incredibly useful for confirming that user settings are optimal before you put in a LocationRequest, however, it isn’t the best approach for continual monitoring. For that, you can use the new LocationCallback class in place of your existing LocationListener to receive LocationAvailability updates in addition to location updates, giving you a simple callback whenever settings might have changed which will affect the current set of LocationRequests. You can also use FusedLocationProviderApi’s getLocationAvailability() to retrieve the current state on demand.

I already switched from the `LocationListener` to the `LocationCallback`. We now must find a way to signal the user that their settings may not be appropriate for what it is requesting. A notification?

**Comment by Bananeweizen** [comment]
Not sure why you want a notification.
```
status.startResolutionForResult( 
                         OuterClass.this, 
                         REQUEST_CHECK_SETTINGS); 
```
That part is meant to invoke the quick change location settings dialog as you may know it from Google Maps. See here for an example screenshot: http://android-developers.blogspot.de/2015/03/google-play-services-70-places-everyone.html

**Comment by samueltardieu** [comment]
Because of `OuterClass.this`.
